snippets/card-product.liquid

```diff
@@ -30,7 +31,7 @@
       assign ratio = 1
     endif
   -%}
-  <div class="product-card-wrapper card-wrapper underline-links-hover">
+  <div class="card-wrapper underline-links-hover {% if wrapper_class == '' %}product-card-wrapper{% else %}{{ wrapper_class }}{% endif %}">
```

Comment:
This currently breaks product cards everywhere but in the sections which are passing `wraper_class` attribute.
---
If you go to a collection page, for instance, the product cards are not showing any image because they're missing `product-card-wrapper` class:
<img width="1143" alt="image" src="https://user-images.githubusercontent.com/37168033/181632773-697ac052-c69f-4ca1-9719-68be8a5b3720.png">

If no `wrapper_class` attribute is passed to this snippet, it won't match `if wrapper_class == ''` condition because `''` isn't the same as "nil" (`''` is a truthy value, btw, everything other than `nil` and `false` is truthy in Liquid). Therefore,  `{{ wrapper_class }}` will render nothing.

Like I mentioned in my previous review, I would suggest revisiting this approach of using `''` as the value of the setting because it can create confusion. The current default state for this snippet is to render `{{ wrapper_class }}` when no `wrapper_class` is passed, and not `product-card-wrapper` which is what we expect.

It's much easier to read something like this:
`{% if wrapper_class != 'none' %}{{ wrapper_class }}{% else %}product-card-wrapper{% endif %}`
and understand that:
- If nothing is passed, it will render `product-card-wrapper` (default attribute value)
- If `none` is passed, it will render `product-card-wrapper` because that is the default style for this component.
- If an override is passed, it will render whatever the value of `wrapper_class` is.

When I see a check for a `''`, I have to actually stop and remember that `''` is truthy in Liquid, but falsy in JavaScript.

I just wrote a lot of information at once so let me know if it's confusing and I can clarify my comments.

Reply:
I see what you mean 🤔 Good catch, Lucas. I just realized that in the footer example there is no `if` statement so even though it may be passing an empty class (''), it is not an issue since we have no `if` check.

I think instead of passing 'none', which may read as a confusing class we could pass the correct class already so we strip down the logic

Let me know your thoughts
https://github.com/Shopify/dawn/pull/1834/commits/f7be18bb6af07f9c62a97daf2c98ed211600414f
https://github.com/Shopify/dawn/pull/1834/commits/a3b89003e6653b28f93d9f2f613b6f53c62ab7ba